### PR TITLE
feat(nrqlcondition): add new field ignore_on_expected_termination

### PR DESCRIPTION
**Pull request by corinnekunze**
### A Note for Maintainers
⚠️ **DO NOT MERGE** this PR, until the changes in the New Relic Go Client PR linked below are merged and released
est. 12 Aug 2024 8AM PT

### Description
https://new-relic.atlassian.net/browse/NR-270144

Adds a new field `ignore_on_expected_termination` to nrql conditions. Currently blocked by the release of it in nerdgraph.

See the Go Client PR below for additional context on this field on NerdGraph.

### PR with relevant Go Client changes
https://github.com/newrelic/newrelic-client-go/pull/1180/files

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic/blob/main/CONTRIBUTING.md#testing) for instructions on running tests locally.

## How to test this change?
blocked by merging to nerdgraph

### CODE REVIEW

File: newrelic/resource_newrelic_nrql_alert_condition.go [hunk 1]
```diff
@@ -261,6 +261,11 @@ func resourceNewRelicNrqlAlertCondition() *schema.Resource {
 				Description:  "The amount of time (in seconds) to wait before considering the signal expired.  Must be in the range of 30 to 172800 (inclusive)",
 				ValidateFunc: validation.IntBetween(30, 172800),
 			},
+			"ignore_on_expected_termination": {
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Description:  "Do not open a loss of signal on an expected termination.",
```
Comment (akane0915) [inline newrelic/resource_newrelic_nrql_alert_condition.go:267]: Nit: `Whether to ignore expected termination of a signal when considering whether to create a loss of signal incident.` Aligns more closely with AGQL description.

**Review by akane0915 — APPROVED** [review]
Approved with a couple nits. I manually tested and it works as expected. Great work!

**Comment by pranav-new-relic** [comment]
FYI folks, the compile seems to be failing because this PR isn't pointing to the relevant go-client changes (making this note for anyone from my team who might take a look at this PR)

**Review by akane0915 — DISMISSED** [review]
Re-approving

**Review by pranav-new-relic — APPROVED** [review]

**Comment by pranav-new-relic** [comment]
attempting to fix lint issues 

image

**Review by pranav-new-relic — APPROVED** [review]

**Pull request merged**